src/jdk.incubator.vector/share/classes/jdk/incubator/vector/AbstractSpecies.java

```diff
@@ -645,7 +645,8 @@ AbstractSpecies<?> computeSpecies(LaneType laneType,
             // bootstrapping.
             throw new AssertionError("bootstrap problem");
         }
-        assert(s.laneType == laneType) : s + "!=" + laneType;
+        // Let the assert condition true for Halffloat
+        assert(s.laneType == laneType) || laneType.switchKey == LaneType.SK_HALFFLOAT  : s + "!=" + laneType;
```

Comment:
Is it better if:

```
if (laneType.switchKey != LaneType.SK_HALFFLOAT) {
   assert(s.laneType == laneType) : s + "!=" + laneType;
}
```

Reply:
With this approach an extra check will be required for each lane type, which can be avoided by OR short circuiting.